### ISSUE TITLE
Update OpenBLAS with Ryzen 2 support

### ISSUE THREAD

**Issue by gcp** [leela-zero/leela-zero#1648]
The official binaries use a compile from OpenBLAS git master to get Ryzen support (the packages in Nuget are very outdated).

I noticed that it does not support Ryzen 2, which was added to OpenBLAS recently. We should update them before the next release.

**Comment by gcp** [comment]
https://sjeng.org/ftp/libopenblas.dll (64 bit)

Putting this in place of the existing one should work.

**Comment by 22nsuk** [comment]
Putting the file in the latest next branch sometimes causes a freezing problem.
My computer is running ryzen 1800x (not ryzen2) + 1080 and the latest version of sabaki on Windows 10 64 bit.
Symptoms appear to have no response while generating genmove.
The strange thing is that the same symptom did not occur in the fp16 version of the same branch.
When I tried building again, my symptoms were repeated.
The problem was resolved by putting the file into the existing one.
I have no idea why this is happening, but it happened.

**Comment by gcp** [comment]
>did not occur in the fp16 version of the same branch.

What does this mean?

You imply that you're making your own builds. Are you able to attach+break into them with a debugger when they hang?

**Comment by 22nsuk** [comment]
1. I have confirmed that the same problem also occurs in the official branch.(another system. fx8300 + R9 380)
2. Only shutdown is allowed (normal shutdown, not forced shutdown).
3. I didn't download it from nuget. I used a dll that came after building with msvc.
Not sure, but it looks the same as the 2015.04.07 version.
4. Now the question is, why doesn't it only happen in the fp16 version of the latest next branch...

**Comment by gcp** [comment]
>I have confirmed that the same problem also occurs in the official branch

What do you mean with "official branch"? I am talking about the release (packages). Not your own build from source.

>I didn't download it from nuget. I used a dll that came after building with msvc.

That's Nuget.

**Comment by 22nsuk** [comment]
@gcp The "official branch" I mean is the leela-zero master branch (0.15 version)
The same symptoms also appeared in 0.15 + new libopenblas.
old libopenblas (nuget?) does not cause any problems.

**Comment by gcp** [comment]
I am still not clear whether you are using the release binaries or building yourself?

**Comment by 22nsuk** [comment]
The two next branches were built by me (fp32, fp16)
Version 0.15 used the release version.

new libopenblas is a ryzen2 supported version that comes here.
old libopenblas means msvc comes out after building (you called nuget).

**Comment by gcp** [comment]
OK. I was asking because the binary is built against the OpenBLAS DLL import lib. The release used mine with a similar config as the the DLL that I uploaded here. If you use the MSVC project file you'll get the Nuget one which might be very different.

I don't really see any way fp16 could make a difference.

**Comment by 22nsuk** [comment]
I'm Beginner, so I finally understood. maybe?
I've done additional testing.
I am not good at English, and I will mark it with symbols.

leelaz.exe
A. 0.15 official release
B. next branch (FP32) / The version I built.
C. next branch (FP16) / The version I built.

libopenblas
1. Included in the 0.15 official release
2. libopenblas after msvc build (nuget)
3. This post's libopenblas (for ryzen2)

A-1: no problem
A-2: no problem
A-3: freeze
B-1: no problem
B-2: no problem
B-3: freeze
C-3: no problem - I have no idea why this works.
